File: CS/DXWebApplication_ReportDesigner/DAL/SessionFactory.cs
using DevExpress.Xpo;
using DevExpress.Xpo.Metadata;
using System.Web.Configuration;

namespace DXWebApplication_ReportDesigner.DAL {
    public static class SessionFactory {
        static readonly IDataLayer dataLayer;

        static SessionFactory() {
            var dictionary = new ReflectionDictionary();
            dictionary.GetDataStoreSchema(typeof(SessionFactory).Assembly);

            var connectionString = WebConfigurationManager.ConnectionStrings["DefaultConnection"].ConnectionString;
            var dataStore = XpoDefault.GetConnectionProvider(connectionString, DevExpress.Xpo.DB.AutoCreateOption.DatabaseAndSchema);

            dataLayer = new ThreadSafeDataLayer(dictionary, dataStore);
        }

        public static UnitOfWork Create() {
            return new UnitOfWork(dataLayer);
        }
    }
}

File: VB/DXWebApplication_ReportDesigner/DAL/SessionFactory.vb
Imports DevExpress.Xpo
Imports DevExpress.Xpo.Metadata
Imports System.Web.Configuration

Namespace DXWebApplication_ReportDesigner.DAL
	Public NotInheritable Class SessionFactory

		Private Sub New()
		End Sub
		Private Shared ReadOnly dataLayer As IDataLayer

		Shared Sub New()
			Dim dictionary = New ReflectionDictionary()
			dictionary.GetDataStoreSchema(GetType(SessionFactory).Assembly)

			Dim connectionString = WebConfigurationManager.ConnectionStrings("DefaultConnection").ConnectionString
			Dim dataStore = XpoDefault.GetConnectionProvider(connectionString, DevExpress.Xpo.DB.AutoCreateOption.DatabaseAndSchema)

			dataLayer = New ThreadSafeDataLayer(dictionary, dataStore)
		End Sub

		Public Shared Function Create() As UnitOfWork
			Return New UnitOfWork(dataLayer)
		End Function
	End Class
End Namespace